### PR TITLE
Added new auth flow

### PR DESCRIPTION
**Pull request by j-rewerts**
- added client linting
 - added a ProtectedRoute
 - normalised frontend style

The main part of this PR is our new auth flow.

1. User visits Mailman.
2. SignalR connects.
3. Request is sent to `/api/user/me` to get the current user. Either they get their user (A) or they aren't authenticated (B)
**A.** 
    4. User object is added to Redux. All server side routes the require auth should work.
**B.** 
    4. 401 indicates the user is unauthenticated. Login flow starts.
    5. Window is opened to `/api/login?SignalRId=whatever`.
    6. If they need to give perms, they're redirected to Google.
    7. When auth completes, using the signalr Id, we notify the user.

### CODE REVIEW

**Comment by j-rewerts** [comment]
@dchenier This PR is probably impossible to review, since I reformatted all of the frontend files. Sorry! 😄

File: src/Mailman.Server/ClientApp/src/components/FetchData.js [hunk 1]
```diff
@@ -1,37 +1,42 @@
-import React, { Component } from 'react';
```
Comment (fizban3) [inline src/Mailman.Server/ClientApp/src/components/FetchData.js]: I think this file can be deleted now :)

Comment (dchenier) [inline src/Mailman.Server/ClientApp/src/components/FetchData.js:1]: I think we can get rid of this file now :)

File: src/Mailman.Server/ClientApp/src/components/create-merge/Date.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  const second = new Date()
+    .getSeconds()
+    .toString()
+    .padStart(2, '0')
+  return (
+    year + '-' + month + '-' + day + ' ' + hour + ':' + minute + ':' + second
+  )
+}
+
+export function getUtcDateString() {
```
Comment (dchenier) [inline src/Mailman.Server/ClientApp/src/components/create-merge/Date.js:28]: Is there an easier way to do this?
(i.e. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString)?

File: src/Mailman.Server/ClientApp/src/components/create-merge/EmailSubject.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        'Content-Type': 'application/json',
+      },
+    }
+
+    const ssid = this.props.spreadsheetId
+    const sheetName = this.props.currentMergeTemplate.sheetName
+    const rowNumber = this.props.currentMergeTemplate.headerRowNumber
+
+    fetch(
+      `https://localhost:5001/api/Sheets/RowValues/${ssid}/${sheetName}?rowNumber=${rowNumber}`,
```
Comment (dchenier) [inline src/Mailman.Server/ClientApp/src/components/create-merge/EmailSubject.js:38]: Any way to avoid hardcoding urls?

**Review by dchenier — COMMENTED** [review]
I know the hardcoded urls were there before this PR, but can we fix them before merging in the code?

**Review by dchenier — APPROVED** [review]

**Comment by j-rewerts** [comment]
Mind if I take a look at those hard coded URLs in the next PR?

**Pull request merged**